lib/blurb/campaign.rb

```diff
@@ -1,29 +1,30 @@
 module Blurb
   class Campaign < BaseResource
-    def self.retrieve(campaign_id)
-      get_request("/v1/campaigns/#{campaign_id}")
+    def self.retrieve(campaign_id, campaign_type)
+      get_request("/v2/#{campaign_type}/campaigns/#{campaign_id}")
     end
 
     def self.retrieve_extended(campaign_id)
-      get_request("/v1/campaigns/extended/#{campaign_id}")
+      get_request("/v2/#{campaign_type}/campaigns/extended/#{campaign_id}")
     end
 
-    def self.list(params = {}, opts = {})
-      get_request("/v1/campaigns?#{setup_url_params(params)}")
+    def self.list(campaign_type, params = {}, opts = {})
+      get_request("/v2/#{campaign_type}/campaigns?#{setup_url_params(params)}")
     end
 
     def self.create(params = {}, opts = {})
       # required argument checks
-      if !params["name"] && !params["campaignType"] && !params["targetingType"] && !params["state"] && !params["dailyBudget"] && !params["startDate"]
-        raise ArgumentError.new("params hash must contain name, campaignType, targetingType, state, dailyBudget and startDate")
+      if !params["name"] && !params["targetingType"] && !params["state"] && !params["dailyBudget"] && !params["startDate"]
+        raise ArgumentError.new("params hash must contain name, targetingType, state, dailyBudget and startDate")
       end
 
-      post_request("/v1/campaigns", [params])
+      post_request("/v2/sp/campaigns", [params])
```

Comment:
Why does this create not take a `campaign_type` parameter, and instead always post to `/v2/sp/campaigns`?

Reply:
Great Question.  The API does not support the creation of sponsored brands (formerly known as headline search ads) through the api.  Sponsored Brands must be created through the user interface so the create method can only be used for sponsored products.

Reply:
Is this something then, where we could keep the API consistent, but throw an exception on our end if an invalid value is passed? That way it wouldn't be breaking if they ever ended up supporting it, or more, different types.